pkg/cli/init.go

```diff
@@ -100,18 +101,24 @@ func (c cli) getAvailableProjectVersions() (projectVersions []string) {
 	for version := range versionSet {
 		projectVersions = append(projectVersions, strconv.Quote(version))
 	}
+	sort.Strings(projectVersions)
 	return projectVersions
 }
 
 func (c cli) getAvailablePlugins() (pluginKeys []string) {
+	keySet := make(map[string]struct{})
 	for _, versionedPlugins := range c.pluginsFromOptions {
 		for _, p := range versionedPlugins {
 			// Only return non-deprecated plugins.
 			if _, isDeprecated := p.(plugin.Deprecated); !isDeprecated {
-				pluginKeys = append(pluginKeys, strconv.Quote(plugin.KeyFor(p)))
+				keySet[plugin.KeyFor(p)] = struct{}{}
```

Comment:
I can see that we're doing `sort.Strings(pluginKeys)` which orders the plugin keys but any reason for building the `keySet` here and then appending the keys to `pluginKeys`?
`pluginKeys = append(pluginKeys, strconv.Quote(plugin.KeyFor(p)))` isn't too unreadable imo.

Reply:
`c.pluginsFromOptions` are indexed by project version, so two `versionedPlugins` slices can be intersecting sets if one plugin supports two project versions (`go/v2` supports project versions `2` and `3-alpha`).

Reply:
Ah, double counting. Gotcha.